File: cpp/n234_isPalindrome.cpp
// https://leetcode-cn.com/problems/palindrome-linked-list/
#include <iostream>
#include <vector>
using namespace std;

// Definition for singly-linked list.
struct ListNode {
    int val;
    ListNode *next;
    ListNode(int x) : val(x), next(NULL) {}
};

class Solution {
public:
    bool isPalindrome(ListNode* head) {
        if (head == NULL) return true;
        ListNode* tmp = head;
        vector<int> mylist;
        while (tmp != NULL){
            mylist.push_back(tmp -> val);
            tmp = tmp -> next;
        }
        int left = 0;
        int right = mylist.size() - 1;
        while (left < right){
            if (mylist[left] != mylist[right]) return false;
            left += 1;
            right -= 1;
        }
        return true;
    }
};

int main(){
    return 0;
}

File: python/n234_isPalindrome.py
# https://leetcode-cn.com/problems/palindrome-linked-list/

# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    def isPalindrome(self, head: ListNode) -> bool:
        if head is None: return True
        mystack = []
        iter = head
        while iter is not None:
            mystack.append(iter.val)
            iter = iter.next
        mystr = ''.join(map(str(), mystack))
        for i in range(0, len(mystack)//2):
            if mystack[i] != mystack[len(mystack)-i-1]:
                return False


if __name__ == "__main__":
    pass        